modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -423,29 +423,33 @@ public void setMuleContext(MuleContext context) {
-                                                                                      (CheckedSupplier<Object>) () -> context
-                                                                                          .getConfig().orElse(null),
-                                                                                      context.getConnectionProvider()
-                                                                                          .orElse(null))));
+      return runWithResolvingContext(context -> withContextClassLoader(classLoader, () -> getValueProviderMediator().getValues(
+                                                                                                                               parameterName,
+                                                                                                                               getParameterValueResolver(),
+                                                                                                                               (CheckedSupplier<Object>) () -> context
+                                                                                                                                   .getConnection()
+                                                                                                                                   .orElse(null),
+                                                                                                                               (CheckedSupplier<Object>) () -> context
+                                                                                                                                   .getConfig()
+                                                                                                                                   .orElse(null),
+                                                                                                                               context
+                                                                                                                                   .getConnectionProvider()
+                                                                                                                                   .orElse(null))));
     } catch (MuleRuntimeException e) {
       Throwable rootException = getRootException(e);
       if (rootException instanceof ValueResolvingException) {
-        throw (ValueResolvingException) rootException;
+        throw (org.mule.runtime.extension.api.values.ValueResolvingException) rootException;
       } else {
-        throw new ValueResolvingException("An unknown error occurred trying to resolve values. " + e.getCause().getMessage(),
-                                          UNKNOWN, e);
+        throw new org.mule.runtime.extension.api.values.ValueResolvingException("An unknown error occurred trying to resolve values. "
+            + e.getCause().getMessage(),
```

Comment:
No, it's checking that there is a known ValueResolvingException as a root cause and throwing UNKNOWN otherwise.

Reply:
but if the root exception is a `org.mule.sdk.api.values.ValueResolvingException` doesn't it get wrapped as an UNKNOWN legacy exception and the true error code gets lost?

Reply:
Additionally, this shouldn't be throwing a `org.mule.runtime.extension.api.values.ValueResolvingException` but a `org.mule.sdk.api.values.ValueResolvingException`. We can't however make that change as tooling is using non API components and then catching for that specific exception. Add a TODO here also pointing to the jira you created explaining that this change needs to happen

Reply:
`getValues` from  `ValueProviderMediator`  throws the legacy exception I can't change that since it's also used by tooling, so it's not possible to get  `org.mule.sdk.api.values.ValueResolvingException` here. The `ValueProviderMediator` catches the SDK exception and throws the legacy exception.

Regarding the second comment besides tooling using the method, this is defined on the ComponentValueProvider API so we should actually deprecate the method and create a new one.